### ISSUE TITLE
Define new tech stack, and stick to it

### ISSUE THREAD

**Issue by duranmla** [hashlabs/guides#8]
The idea is to keep track of those things that we thing may work in order to define the final tech stack. we need to define things like:
- [ ] Front-End Stack 
- [ ] Back-End Stack
- [ ] DevOps Stack
- [ ] Testing tools

> Don't hesitate to add more deep description to this issue in order to be able to keep the conversation context and mark every accomplish decision

**Comment by orlando** [comment]
# Mobile development stack
- If we don't need hardcore features like OpenGL or Metal. We will use React-native for all of our mobile development

# Front-end Stack
- SCSS
- Flexbox (Bootstrap 4 has flexbox support, but we have to check it. There's also http://flexboxgrid.com/ which looks really good). Maybe @castrolem can help with this one.
- https://github.com/10up/flexibility for ie backwards compatiblity
- React

# Back-end Stack
- Ruby on Rails for basic projects
- Golang for microservices
- Postgres as our main database
- Redis as our cache database
- Other NoSQL database where's needed
- We need also to consider a framework that allows us prototype/build application quickly. This may have more sense in a mobile application, but we can use it for desktop applications too depending the case. Something like:
  - **Parse Server**
    - Cool and works out of the box, has a really useful dashboard that non technical people can use.
  - **Horizon.io**
    - Backed by RethinkDB which looks like a really good NoSQL database. Has no dashboard right now but is in the works (https://github.com/rethinkdb/horizon/issues/154)

# DevOps stack
- [Docker](http://www.docker.com) for everything, including local development. You should have by now a Docker beta invite, if not request one at https://beta.docker.com/
- [Rancher](http://www.rancher.com) for containers management
- [Prometheus](http://www.prometheus.io) for service monitoring
- [ELK](https://www.elastic.co/products) for logging analysis
- [Sentry](http://getsentry.com) for our error reporting
- DigitalOcean to deploy or AWS (I prefer digitalocean since it's cheaper)

# Testing Tools
- Our CI/CD process will be managed using drone.io, We will have an instance running and every repo has to have Drone configured
- Testing framework are language/framework dependant. For Rails we will use what Rueditas has. For Golang we will use the default testing package. For Node.js/React we will use Jest or Mocha (not one defined here, so let's play with both and see how it goes

**Comment by duranmla** [comment]
Awesome! Getting everything into shape! 💯

**Comment by duranmla** [comment]
We should mark this goal on Lattice, cause we have already make some decisions, besides, the only current thing to change will be use Firebase instead Parse

**Comment by orlando** [comment]
Yeah, also we need to change other things in here, let's talk about this tomorrow
